### PR TITLE
AlignWithClosestGridTile extension for EntityCoordinates

### PR DESCRIPTION
**Pull request by gradientvera**
Used for aligning an EntityCoordinates to the closest grid. Does the same as the `AlignTileAny` placement mode.
Also fixes component exception when adding a timer component to a deleted entity.

### CODE REVIEW

File: Robust.Shared/Map/CoordinatesExtensions.cs [hunk 1]
```diff
@@ -12,8 +13,64 @@ public static EntityCoordinates ToEntityCoordinates(this Vector2i vector, GridId
...
+        {
+            var coords = coordinates;
+            entityManager ??= IoCManager.Resolve<IEntityManager>();
+            mapManager ??= IoCManager.Resolve<IMapManager>();
+
+            var gridId = coords.GetGridId(entityManager);
+
+            IMapGrid? mapGrid = null;
+
+            if (!gridId.IsValid() || !mapManager.TryGetGrid(gridId, out mapGrid))
```
Comment (metalgearsloth) [inline Robust.Shared/Map/CoordinatesExtensions.cs]: Couldn't you just use out var mapGrid?

**Review by metalgearsloth — APPROVED** [review]
LGTM considering it's a port.

File: Robust.Shared/Map/CoordinatesExtensions.cs (hunk 1 above)
Comment (gradientvera) [reply Robust.Shared/Map/CoordinatesExtensions.cs]: Done. It was also unused so I renamed it to `_`

Comment (gradientvera) [reply Robust.Shared/Map/CoordinatesExtensions.cs]: Actually, given how it's unused I just changed this to `GridExists()`

**Pull request merged**